ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/P2PDumpManager.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class P2PDumpManager {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String GOSSIP_MESSAGES_DIR = "gossip_messages";
+  private static final String DECODING_ERROR_SUB_DIR = "decoding_error";
+  private static final String REJECTED_SUB_DIR = "rejected";
+  private static final String INVALID_BLOCK_DIR = "invalid_blocks";
+
+  private final boolean enabled;
+  private final Path directory;
+
+  public P2PDumpManager(final Path directory, final boolean enabled) {
+    this.enabled = enabled;
+    this.directory = directory;
+    if (!enabled) {
+      return;
+    }
+
+    final Path gossipMessagesPath = this.directory.resolve(GOSSIP_MESSAGES_DIR);
+    if (gossipMessagesPath.toFile().mkdirs()) {
+      LOG.info("{} directory has been created to save gossip messages.", GOSSIP_MESSAGES_DIR);
+    }
+    if (gossipMessagesPath.resolve(DECODING_ERROR_SUB_DIR).toFile().mkdirs()) {
+      LOG.info(
+          "{} directory has been created to save gossip messages with decoding errors.",
+          DECODING_ERROR_SUB_DIR);
+    }
+    if (gossipMessagesPath.resolve(REJECTED_SUB_DIR).toFile().mkdirs()) {
+      LOG.info(
+          "{}} directory has been created to save rejected gossip messages.", REJECTED_SUB_DIR);
+    }
+    if (this.directory.resolve(INVALID_BLOCK_DIR).toFile().mkdirs()) {
+      LOG.info(
+          String.format(
+              "%s directory has been created to save invalid blocks.", INVALID_BLOCK_DIR));
+    }
```

Comment:
If we fail to create these subdirectories, we are gonna keep trying to write the messages to them later. I think we should think of a circuit breaker mechanism to ensure we don't waste time trying to write to a directory that we know doesn't exit.

Reply:
all of these creates i'd be making consistent, and also i'd be using log.debug.
consistency in terms of always using log.info, we dont need to have the last one using string.format for no reason...